ydb/library/yql/public/udf/udf_data_type.cpp

```diff
@@ -23,52 +23,55 @@ ui8 GetDecimalWidth(EDataTypeFeatures features, size_t size) {
+    {{ NO,  MF, MF, MF, MF, MF, MF, OK, OK,  LD, LD,  OK, OK,  NO, NO,  NO,  LD, LD, OK,  NO,  LD, LD, OK,  NO, NO, NO,  NO, NO, NO,  NO,  LD, LD, OK }}, // TzTimestamp
 
-    {{ NO,  UN, UN, UN, UN, UN, UN, UN, UN,  LD, LD,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  UN, NO, NO }}, // Decimal
+    {{ NO,  UN, UN, UN, UN, UN, UN, UN, UN,  LD, LD,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  UN, NO, NO,  NO, NO, NO,  NO,  NO, NO, NO }}, // Decimal
+    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  NO, OK, NO,  NO, NO, NO,  NO,  NO, NO, NO }}, // DyNumber
+    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, OK,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  NO, NO, OK,  NO, NO, NO,  NO,  NO, NO, NO }}, // JsonDocument
 
-    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  NO, OK, NO }}, // DyNumber
+    {{ NO,  MF, MF, MF, MF, OK, MF, OK, MF,  LD, OK,  OK, OK,  NO, NO,  NO,  MF, MF, MF,  NO,  NO, NO, NO,  NO, NO, NO,  OK, OK, OK,  NO,  NO, NO, NO }}, // Date32
+    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  FL, MF, MF,  NO,  NO, NO, NO,  NO, NO, NO,  LD, OK, OK,  NO,  NO, NO, NO }}, // Datetime64
+    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  FL, FL, MF,  NO,  NO, NO, NO,  NO, NO, NO,  LD, LD, OK,  NO,  NO, NO, NO }}, // Timestamp64
 
-    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, OK,  NO,  NO, NO, NO,  NO,  NO, NO, NO,  NO, NO, OK }}, // JsonDocument
+    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  MF,  NO, NO, NO,  NO, NO, NO,  NO, NO, NO,  OK,  NO, NO, NO }}, // Interval64
 
-    {{ NO,  MF, MF, MF, MF, OK, MF, OK, MF,  LD, OK,  OK, OK,  NO, NO,  NO,  MF, MF, MF,  NO,  NO, NO, NO,  NO, NO, NO, OK, OK, OK, NO }}, // Date32
-    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  FL, MF, MF,  NO,  NO, NO, NO,  NO, NO, NO, LD, OK, OK, NO }}, // Datetime64
-    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  FL, FL, MF,  NO,  NO, NO, NO,  NO, NO, NO, LD, LD, OK, NO }}, // Timestamp64
-    {{ NO,  MF, MF, MF, MF, MF, MF, OK, MF,  LD, LD,  OK, OK,  NO, NO,  NO,  NO, NO, NO,  MF,  NO, NO, NO,  NO, NO, NO, NO, NO, NO, OK }}, // Interval64
+    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, NO,  NO,  MF, MF, MF,  NO,  MF, MF, MF,  NO, NO, NO,  OK, OK, OK,  NO,  OK, OK, OK }}, // TzDate32
+    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, NO,  NO,  MF, MF, MF,  NO,  MF, MF, MF,  NO, NO, NO,  LD, OK, OK,  NO,  LD, OK, OK }}, // TzDatetime64
+    {{ NO,  NO, NO, NO, NO, NO, NO, NO, NO,  NO, NO,  OK, OK,  NO, NO,  NO,  MF, MF, MF,  NO,  MF, MF, MF,  NO, NO, NO,  LD, LD, OK,  NO,  LD, LD, OK }}, // TzTimestamp64
 }};
 
 }
```

Comment:
Considering the changes over the sources, casts mapping update and the introduced ut, validating no holes are in casts map, I believe there is no need in `TMaybe` for `GetCastResult` return type (as well as no reason to return `Nothing()` if the particular cast is missing. What about adding `Y_ENSURE` there instead?

Reply:
It's a good idea. I could integrate it into following PRs